packages/editor/src/components/browser/tableCol/TableColBrowser.tsx

```diff
@@ -1,21 +1,136 @@
-import { useState } from 'react';
-import { Input } from '../../widgets';
+import { useEffect, useMemo, useState } from 'react';
+import { SelectRow, Table, TableCell } from '../../widgets';
 import type { UseBrowserImplReturnValue } from '../useBrowser';
 import { IvyIcons } from '@axonivy/editor-icons';
 import type { BrowserValue } from '../Browser';
+import { useEditorContext, useMeta } from '../../../context';
+import {
+  useReactTable,
+  type ColumnDef,
+  type RowSelectionState,
+  getCoreRowModel,
+  getFilteredRowModel,
+  flexRender
+} from '@tanstack/react-table';
+import { useQueryData } from '../../parts/query/useQueryData';
+import type { Column } from '../../parts/query/database/TableReadFields';
 export const TABLE_COL_BROWSER_ID = 'tablecol' as const;
 
-export const useTableColBrowser = (): UseBrowserImplReturnValue => {
-  const [value, setValue] = useState<BrowserValue>({ cursorValue: 'table column' });
+export const useTableColBrowser = (onDoubleClick: () => void): UseBrowserImplReturnValue => {
+  const [value, setValue] = useState<BrowserValue>({ cursorValue: '' });
   return {
     id: TABLE_COL_BROWSER_ID,
     name: 'Table Column',
-    content: <TableColumnBrowser value={value.cursorValue} onChange={(change: string) => setValue({ cursorValue: change })} />,
+    content: <TableColumnBrowser value={value.cursorValue} onChange={setValue} onDoubleClick={onDoubleClick} />,
     accept: () => value,
     icon: IvyIcons.Rule
   };
 };
 
-const TableColumnBrowser = (props: { value: string; onChange: (value: string) => void }) => {
-  return <Input {...props} />;
+const TableColumnBrowser = (props: { value: string; onChange: (value: BrowserValue) => void; onDoubleClick: () => void }) => {
+  const { elementContext: context } = useEditorContext();
+  const { config } = useQueryData();
+
+  const columnMetas = useMeta('meta/database/columns', { context, database: config.query.dbName, table: config.query.sql.table }, []).data;
+
+  const [data, setData] = useState<Column[]>([]);
+
+  const [showHelper, setShowHelper] = useState(false);
+
+  useEffect(() => {
+    const select = config.query.sql.select;
+    console.log(columnMetas);
+    console.log(select);
+
+    const columnData = columnMetas.map<Column>(c => {
+      return { ...c, selected: select.includes(c.name) };
+    });
+    console.log(columnData);
+    setData(columnData.filter(c => c.selected === true));
```

Comment:
why split this map and filter? do in one step:
```ts
columnMetas.filter(c => c.select.includes(c.name));
```

or why is this selected needed?

Reply:
does this works too if you select all rows via the checkbox?

Reply:
Well, I've written the code more compactly now, but I have to map it first and then filter it because I don't receive the 'select' information directly from the metas but instead through the useQueryData hook.

And you're right, that didn't work with the checkbox. I've adjusted it now so that it works with the checkbox as well.